### PR TITLE
Feature/docker

### PR DESCRIPTION
**Pull request by mathematicalmichael**
this is about as minimal a change as I could manage. 
- not deleting pycache... though you really should, but in a separate PR.
- not deleting `console/static/console` as it was causing confusion. Need to remove the files that are dynamically generated and make sure they are generated during `make install` or something like that (`make node_modules` maybe?)
- appears that changing the light/dark theme is currently broken (on main too)
- `make .env` creates a file if there isn't one from the example file, fills in as much of it as possible
- `make .docker.env` for the docker env file (copied from .env)
- `make audit-fix` has been run, 1 requiring manual review
- `make update` updates things ...
- `make install` builds the image and installs node modules, worth running after audit-fix or `make update`
- `make live` then runs the live server (accessible both locally and remote! the latter is important for people like myself developing on ARM architecture through ssh to a linux box) and the website is visible on port 8081 (can make it 8080 again before merging the PR, I just have apache guacamole running on the same machine on that port already)
- `make live` also runs `npm migrate` as part of its instructions, which differs from before
- there are warnings about the webpack live server migrating to another package or something like that... result of `python manage.py runserver`
- don't think you need the `Dockerfile` anymore. Perhaps rename `Dockerfile-node` to `Dockerfile`? Not sure what your intentions were for the original dockerfile, but this new one has "the whole development stack"

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+build:
+	docker build -t cannlytics -f Dockerfile .
+
+npm:
+	docker run --rm -it \
+	--name npm \
+    -p 8090:8080 \
+    -p 8000:8000 \
+    -p 8081:8081 \
+    -p 35729:35729 \
```
Comment (mathematicalmichael) [inline Makefile:15]: fix spacing

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	npm-image $(c)
+
+
+install: npm_image node_modules
+
+install_firebase_tools:
+	make npm c="npm install -g firebase-tools"
+
+install_webpack:
+	make npm c="npm install webpack-dev-server --save-dev"
```
Comment (mathematicalmichael) [inline Makefile:27]: unsure if this is required.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+webpack-dev:
+	make npm c="webpack-dev-server --env production=False"
```
Comment (mathematicalmichael) [inline Makefile:30]: dont think this is needed

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+.PHONY: node_modules
+node_modules:
+	make npm c="npm install"
+
+live:
```
Comment (mathematicalmichael) [inline Makefile:36]:
```suggestion
start:
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+docker.env:
+	cp .env docker.env
+
+.env:
+	cp .env.example .env
+	sed -i '/SECRET_KEY/d' .env
+	docker run --rm -it \
+	-v `pwd`/django_secret.py:/app/django_secret.py \
+	cannlytics-back \
```
Comment (mathematicalmichael) [inline Makefile:53]:
```suggestion
	npm-image \
```

**Review by keeganskeate — APPROVED** [review]
Awesome work Michael!

Just for future reference, these are the resources that I used to use a Makefile on Windows.

- [Download MinGW](https://stackoverflow.com/a/62450734/5021266)
- [Makefile Tutorial](https://makefiletutorial.com/)

The makefile can then be ran with the following command from the `cannlytics` root directory (assuming that MinGW is installed at `C:/MinGW`):

```
C:/MinGW/bin/mingw32-make.exe -f makefile.mak
```

**Pull request merged**